sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -268,13 +292,23 @@ private[parquet] object ParquetReadSupport {
    * @return A list of clipped [[GroupType]] fields, which can be empty.
    */
   private def clipParquetGroupFields(
-      parquetRecord: GroupType, structType: StructType): Seq[Type] = {
-    val parquetFieldMap = parquetRecord.getFields.asScala.map(f => f.getName -> f).toMap
+      parquetRecord: GroupType,
+      structType: StructType,
+      caseInsensitive: Boolean): Seq[Type] = {
+    val parquetFieldMap = {
+      val pairs = parquetRecord.getFields.asScala.map(f => f.getName -> f)
+      implicit val ordering = if (caseInsensitive) {
+        Ordering.by[String, String](_.toLowerCase)
+      } else {
+        Ordering.String
+      }
+      TreeMap(pairs: _*)
```

Comment:
The implicit ordering val determines if the TreeMap behaves in a case-sensitive manner or not. I can rework this into using standard Maps if you feel thats appropriate or clearer.